### PR TITLE
Add frontend support for `date` platform

### PR DESCRIPTION
**Pull request by raman325**
## Proposed change
Adds support for the `date` platform outlined in https://github.com/home-assistant/architecture/discussions/811

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- Link to core pull request: https://github.com/home-assistant/core/pull/81948
- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/24986

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by raman325** [comment]
not stale

File: src/common/entity/compute_state_display.ts [hunk 1]
```diff
@@ -126,7 +126,7 @@ export const computeStateDisplayFromEntityAttributes = (
         );
         return formatDateTime(date, locale);
       }
-      if (attributes.has_date) {
+      if (domain === "date" || attributes.has_date) {
         date = new Date(attributes.year, attributes.month - 1, attributes.day);
```
Comment (frenck) [inline src/common/entity/compute_state_display.ts:130]: The `date` entity will not have these attributes, so, in that case, we need to create a date from the state string.

I don't think this part is even needed for this integration, date parsing done above, seems to be enough?

**Comment by frenck** [comment]
Backend has been merged for 2023.6

File: src/common/entity/compute_state_display.ts (hunk 1 above)
Comment (raman325) [reply src/common/entity/compute_state_display.ts:130]: correct, removing

File: src/panels/lovelace/entity-rows/hui-date-entity-row.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  static get styles(): CSSResultGroup {
+    return css`
+      ha-date-input + ha-time-input {
+        margin-left: 4px;
+        margin-inline-start: 4px;
+        margin-inline-end: initial;
+        direction: var(--direction);
+      }
+    `;
+  }
```
Comment (bramkragten) [inline src/panels/lovelace/entity-rows/hui-date-entity-row.ts:84]:
```suggestion
```

File: src/panels/lovelace/entity-rows/hui-date-entity-row.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+  private _dateChanged(ev: CustomEvent<{ value: string }>): void {
+    const stateObj = this.hass!.states[this._config!.entity];
+
+    setDateValue(this.hass!, stateObj.entity_id, ev.detail.value);
+  }
+
+  static get styles(): CSSResultGroup {
+    return css`
+      ha-date-input + ha-time-input {
```
[the remark this answers is not in the document]
Comment (bramkragten) [reply src/panels/lovelace/entity-rows/hui-date-entity-row.ts:77]: Also in the `time` PR btw

**Review by bramkragten — APPROVED** [review]

**Pull request merged**